scipy/_lib/_testutils.py

```diff
@@ -33,7 +33,7 @@ def __call__(self, label="fast", verbose=1, extra_argv=None, doctests=False,
         module = sys.modules[self.module_name]
         module_path = os.path.abspath(module.__path__[0])
 
-        pytest_args = ['-l']
```

Comment:
I think `-l` is useful, especially for parametrised tests where it makes it easier to see which parameters are failing.

Reply:
`-l` doesn't seem to do anything for `native`/`short` modes, that's why I removed it. Perhaps I only tested `'native'`, not sure - the pytest code [here](https://github.com/pytest-dev/pytest/blob/7e5ad314287a5c86856efea13c04333d7baf3643/src/_pytest/_code/code.py#L586) says:
```
        :param bool showlocals:
            Show locals per traceback entry.
            Ignored if ``style=="native"`
```
I'll try putting it back.

Reply:
Local testing suggests that doing `--showlocals` does not work with `native` or `short` modes.

Perhaps we should do `--tb=short` only on CIs, leaving local tests as they are (don't specify `tb`).

Reply:
I'd prefer not, `auto`/`long` is an abomination. Still missing `nose`, `pytest`'s defaults are so much worse:(

I'd rather file a bug report for this with Pytest, at least it's well-maintained and releases quickly. So probably the `short` + `showlocals` combo will work soon.

Reply:
Agreed it would be best if we could do `short`+`showlocals`

Reply:
Turns out the issue existed already, added a comment and linked this PR.

Reply:
That's unfortunate, Looks like the bug isn't receiving any attention.

Reply:
Could be due to the terrible history in that issue. I'll ping about that.